web/src/main/java/com/bazaarvoice/emodb/web/megabus/MegabusStashModule.java

```diff
@@ -0,0 +1,125 @@
+import static java.util.Objects.requireNonNull;
+
+public class MegabusStashModule extends PrivateModule {
+
+    private final MegabusBootConfiguration _config;
+
+    public MegabusStashModule(MegabusBootConfiguration configuration) {
+        _config = requireNonNull(configuration);
+
+        checkArgument(_config.getScanThreadCount() > 0, "Scan thread count must be at least 1");
+    }
+
+    @Override
+    protected void configure() {
+        binder().requireExplicitBindings();
+
+        bind(ScanUploader.class).asEagerSingleton();
+        bind(ScanStatusDAO.class).to(DataStoreScanStatusDAO.class).asEagerSingleton();
+        bind(RangeScanUploader.class).to(LocalRangeScanUploader.class).asEagerSingleton();
+        bind(ScanWorkflow.class).toProvider(ScanUploadModule.QueueScanWorkflowProvider.class).asEagerSingleton();
+
+        bind(ScanCountListener.class).to(MetricsScanCountListener.class).asEagerSingleton();
+
+        bind(String.class).annotatedWith(ScanStatusTable.class).toInstance(_config.getScanStatusTable());
+        bind(Integer.class).annotatedWith(MaxConcurrentScans.class).toInstance(_config.getScanThreadCount());
+
+        bind(new TypeLiteral<Optional<String>>(){}).annotatedWith(Names.named("pendingScanRangeQueueName"))
+                .toInstance(Optional.of(_config.getPendingScanRangeQueueName()));
+        bind(new TypeLiteral<Optional<String>>(){}).annotatedWith(Names.named("completeScanRangeQueueName"))
+                .toInstance(Optional.of(_config.getCompleteScanRangeQueueName()));
+
+        bind(ScanWriterGenerator.class).to(MegabusScanWriterGenerator.class).asEagerSingleton();
+
+        install(new FactoryModuleBuilder()
+                .implement(KafkaScanWriter.class, KafkaScanWriter.class)
+                .build(KafkaScanWriterFactory.class));
+
+        // Monitors active upload status, active only by leader election
+        bind(ScanUploadMonitor.class).asEagerSingleton();
+        // Monitors for new scans waiting to start
+        bind(DistributedScanRangeMonitor.class).asEagerSingleton();
+
+        bind(MegabusBootDAO.class).to(StashMegabusBootDAO.class).asEagerSingleton();
+        expose(MegabusBootDAO.class);
+    }
+
+    @Provides
+    @Singleton
+    @ScanStatusTablePlacement
+    protected String provideScanStatusTablePlacement(@SystemTablePlacement String tablePlacement) {
+        return tablePlacement;
+    }
+
+    @Provides
+    @Singleton
+    @Named("ScannerAPIKey")
+    protected String provideScannerApiKey(@ServerCluster String cluster) {
+        checkArgument(_config.getQueueServiceApiKey() != null);
+        String scannerApiKey = _config.getQueueServiceApiKey();
+        if (ApiKeyEncryption.isPotentiallyEncryptedApiKey(scannerApiKey)) {
+            scannerApiKey = new ApiKeyEncryption(cluster).decrypt(scannerApiKey);
+        } else {
+            LoggerFactory.getLogger("com.bazaarvoice.emodb.security").warn(
```

Comment:
it's worth to change log level to `error` and create alert in kibana

Reply:
This is set up the same way in `ScanUploadModule`, and I'm of the opinion it should stay this way, especially because we run megabus locally with the key in plain text.

Reply:
Also, I'm sure we would set up an alert with log level `warn`

Reply:
ok, but we still need to set up alerts

Reply:
agreed!

Reply:
need one more ticket from you 😄